#!/usr/bin/env python
PACKAGE='resized_image_transport'

from dynamic_reconfigure.parameter_generator_catkin import *;

from math import pi
from driver_base.msg import SensorLevels

gen = ParameterGenerator()

#gen.add("max_range", double_t, SensorLevels.RECONFIGURE_RUNNING,
#        "max_range corresponding to max sensor output",   5.0, 0.0, 20.0)

gen.add("resize_scale_x", double_t, SensorLevels.RECONFIGURE_RUNNING,
        "resize scale width",  1.0, 0.01, 1.0)
gen.add("resize_scale_y", double_t, SensorLevels.RECONFIGURE_RUNNING,
        "resize scale height",  1.0, 0.01, 1.0)
gen.add("log_polar_scale", double_t, SensorLevels.RECONFIGURE_RUNNING,
        "log polar scale",  100.0, 1.0, 500.0)

gen.add("msg_par_second", double_t, SensorLevels.RECONFIGURE_RUNNING,
        "number of pub image par second", 15.0, 0.01, 30)

gen.add("verbose", bool_t, SensorLevels.RECONFIGURE_RUNNING,
        "verbose output", False)
exit(gen.generate(PACKAGE, "resized_image_transport", "LogPolar"))
